### ISSUE TITLE
DNS servers lack IPv6 support, DNS recursion fails causing connection timeouts.

### ISSUE THREAD

**Issue by bert128** [microsoft/vscode#212682]
Does this issue occur when all extensions are disabled?: No, the functionality used does not exist.

- VS Code Version: 1.89.1
- OS Version: MacOS 14.4.1

Steps to Reproduce:

1. Open a remote location via SSH to a host that only has IPv6 connectivity, including IPv6 DNS resolution
2. Connection hangs for several minutes
3. Debugging shows the host is attempting to resolve Host "update.code.visualstudio.com", which hangs

What's happening is DNS resolution proceeds through several CNAME records:

$ host update.code.visualstudio.com.
update.code.visualstudio.com is an alias for vscode-update-g6gcb6ggdthkctd9.z01.azurefd.net.
vscode-update-g6gcb6ggdthkctd9.z01.azurefd.net is an alias for **star-azurefd-prod.trafficmanager.net.**
star-azurefd-prod.trafficmanager.net is an alias for **shed.dual-low.part-0031.t-0009.t-msedge.net**.
shed.dual-low.part-0031.t-0009.t-msedge.net is an alias for **part-0031.t-0009.t-msedge.net**.
part-0031.t-0009.t-msedge.net has address 13.107.213.59
part-0031.t-0009.t-msedge.net has address 13.107.246.59
part-0031.t-0009.t-msedge.net has IPv6 address 2620:1ec:46::59
part-0031.t-0009.t-msedge.net has IPv6 address 2620:1ec:bdf::59

The DNS servers for visualstudio.net have IPv6, as do the DNS servers for azurefd.net so DNS resolution is successful up to this point. However the DNS servers for "trafficmanager.net" and "t-msedge.net" do not have IPv6, so DNS resolution fails at this point unless your DNS resolver has legacy connectivity. 

Examples:

$ host -t NS t-msedge.net
t-msedge.net name server ns2.t-msedge.net.
t-msedge.net name server ns1.t-msedge.net.
t-msedge.net name server ns4.t-msedge.net.
t-msedge.net name server ns3.t-msedge.net.
$ host ns2.t-msedge.net.
ns2.t-msedge.net has address 13.107.246.2
$ host ns4.t-msedge.net.
ns4.t-msedge.net has address 13.107.213.2

**Comment by rzhao271** [comment]
This sounds like an issue with Azure or our configuration of some Azure components.

@bert128 are you still seeing the issue to this day?

**Comment by bert128** [comment]
Yes, the DNS servers for t-msedge.net still only have legacy A records and no AAAA records, although it looks like MS have fixed trafficmanager.net as that's now ok.

**Comment by bert128** [comment]
Yes, the domain has changed but the problem persists:

```
$ host update.code.visualstudio.com
update.code.visualstudio.com is an alias for vscode-update-g6gcb6ggdthkctd9.z01.azurefd.net.
vscode-update-g6gcb6ggdthkctd9.z01.azurefd.net is an alias for star-azurefd-prod.trafficmanager.net.
star-azurefd-prod.trafficmanager.net is an alias for shed.dual-low.part-0031.t-0009.t-msedge.net.
shed.dual-low.part-0031.t-0009.t-msedge.net is an alias for azurefd-t-fb-prod.trafficmanager.net.
azurefd-t-fb-prod.trafficmanager.net is an alias for dual.part-0031.t-0009.fb-t-msedge.net.
dual.part-0031.t-0009.fb-t-msedge.net is an alias for part-0031.t-0009.fb-t-msedge.net.
part-0031.t-0009.fb-t-msedge.net has address 13.107.253.59
part-0031.t-0009.fb-t-msedge.net has address 13.107.226.59
part-0031.t-0009.fb-t-msedge.net has IPv6 address 2620:1ec:48:1::59
part-0031.t-0009.fb-t-msedge.net has IPv6 address 2620:1ec:29:1::59
$ host -t NS fb-t-msedge.net
fb-t-msedge.net name server ns2.fb-t-msedge.net.
fb-t-msedge.net name server ns4.fb-t-msedge.net.
fb-t-msedge.net name server ns1.fb-t-msedge.net.
fb-t-msedge.net name server ns3.fb-t-msedge.net.
$ host ns4.fb-t-msedge.net.
ns4.fb-t-msedge.net has address 13.107.226.2
```

it now redirects to fb-t-msedge.net instead of t-msedge.net, but this domain also has no AAAA records on its nameservers.

**Comment by tmottabr** [comment]
I had a similar problem. Not with this specific domain, but with many other that resolve to the same endpoint in MS/Azure.

Came across this bug report while researching.

I also found this other post on Netgate forum about a similar problem: https://forum.netgate.com/topic/198586/can-t-connect-to-vscode-server-via-ipv6-from-clients-but-from-pfsense/6

It mention that setting the MSS to 1452 on the router have solved the problem for them.. I made the same configuration in my router at the wan port and and it did not helped, but setting the MTU to that same value appear to have solved it for me.

Hope it helps those with problem and that MS fix this soon.
